hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -192,6 +232,105 @@ singlePartyCommitsFromExternal tracer workDir node@RunningNode{networkId} hydraS
  where
   RunningNode{nodeSocket} = node
 
+-- | Single hydra-node where the commit is done from an external UTxO owned by a
+-- script which requires providing script, datum and redeemer instead of
+-- signing the transaction.
+singlePartyCommitsFromExternalScript ::
+  Tracer IO EndToEndLog ->
+  FilePath ->
+  RunningNode ->
+  TxId ->
+  IO ()
+singlePartyCommitsFromExternalScript tracer workDir node hydraScriptsTxId =
+  (`finally` returnFundsToFaucet tracer node Alice) $ do
+    refuelIfNeeded tracer node Alice 25_000_000
+    aliceChainConfig <- chainConfigFor Alice workDir nodeSocket [] $ UnsafeContestationPeriod 100
+    let hydraNodeId = 1
+    withHydraNode tracer aliceChainConfig workDir hydraNodeId aliceSk [] [1] hydraScriptsTxId $ \n1 -> do
+      send n1 $ input "Init" []
+      headId <- waitMatch 600 n1 $ headIsInitializingWith (Set.fromList [alice])
+
+      let script = fromPlutusScript @PlutusScriptV2 $ Plutus.alwaysSucceedingNAryFunction 2
+          scriptAddress = mkScriptAddress @PlutusScriptV2 networkId script
+      (someVk, someSk) <- generate genKeyPair
+      pparams <- queryProtocolParameters networkId nodeSocket QueryTip
+      normalUTxO <- seedFromFaucet node someVk 10_000_000 Normal (contramap FromFaucet tracer)
+      scriptUtxo <- createScriptOutput pparams scriptAddress someSk normalUTxO
+      colateralUTxO <- seedFromFaucet node someVk 20_000_000 Normal (contramap FromFaucet tracer)
+
+      -- Request to build a draft commit tx from hydra-node
+      let reedemer = toScriptData () -- ScriptDataBytes mempty
+          datum = toScriptData () -- ScriptDataBytes mempty
+          collateralTxIns = fst <$> UTxO.pairs colateralUTxO
+          scriptInfo = ScriptInfo reedemer datum script collateralTxIns
```

Comment:
As we are implementing option A, I think the collateral can come from the hydra-node. Similarly as the utxo to pay fees.